iocore/hostdb/HostDB.cc

```diff
@@ -300,131 +237,77 @@ HostDBProcessor::cache()
+HostDBBackgroundTask::HostDBBackgroundTask(int frequency) : Continuation(new_ProxyMutex()), frequency(frequency), start_time(0)
 {
-  SET_HANDLER(&HostDBSyncer::sync_event);
+  SET_HANDLER(&HostDBBackgroundTask::sync_event);
 }
 
 int
-HostDBSyncer::sync_event(int, void *)
+HostDBBackgroundTask::wait_event(int, void *)
 {
-  SET_HANDLER(&HostDBSyncer::wait_event);
-  start_time = Thread::get_hrtime();
-  hostDBProcessor.cache()->sync_partitions(this);
-  return EVENT_DONE;
-}
+  ink_hrtime next_sync = HRTIME_SECONDS(this->frequency) - (Thread::get_hrtime() - start_time);
 
-int
-HostDBSyncer::wait_event(int, void *)
-{
-  ink_hrtime next_sync = HRTIME_SECONDS(hostdb_sync_frequency) - (Thread::get_hrtime() - start_time);
-
-  SET_HANDLER(&HostDBSyncer::sync_event);
+  SET_HANDLER(&HostDBBackgroundTask::sync_event);
   if (next_sync > HRTIME_MSECONDS(100))
     eventProcessor.schedule_in(this, next_sync, ET_TASK);
   else
     eventProcessor.schedule_imm(this, ET_TASK);
   return EVENT_DONE;
 }
 
+struct HostDBSync : public HostDBBackgroundTask {
+  std::string storage_path;
+  std::string full_path;
+  HostDBSync(int frequency, std::string storage_path, std::string full_path)
+    : HostDBBackgroundTask(frequency), storage_path(storage_path), full_path(full_path){};
+  int
+  sync_event(int, void *)
+  {
+    SET_HANDLER(&HostDBSync::wait_event);
+    start_time = Thread::get_hrtime();
+
+    new RefCountCacheSync<RefCountCache<HostDBInfo>>(this, hostDBProcessor.cache()->refcountcache, this->frequency,
+                                                     this->storage_path, this->full_path);
+    return EVENT_DONE;
+  }
+};
+
 int
 HostDBCache::start(int flags)
 {
-  Store *hostDBStore;
-  Span *hostDBSpan;
+  (void)flags; // unused
   char storage_path[PATH_NAME_MAX];
-  int storage_size = 33554432; // 32MB default
-
-  bool reconfigure = ((flags & PROCESSOR_RECONFIGURE) ? true : false);
-  bool fix         = ((flags & PROCESSOR_FIX) ? true : false);
+  int hostdb_max_size   = 33554432; // 32MB default
+  int hostdb_partitions = 64;
 
   storage_path[0] = '\0';
 
   // Read configuration
   // Command line overrides manager configuration.
   //
   REC_ReadConfigInt32(hostdb_enable, "proxy.config.hostdb");
-  REC_ReadConfigString(hostdb_filename, "proxy.config.hostdb.filename", sizeof(hostdb_filename));
-  REC_ReadConfigInt32(hostdb_size, "proxy.config.hostdb.size");
   REC_ReadConfigInt32(hostdb_srv_enabled, "proxy.config.srv_enabled");
   REC_ReadConfigString(storage_path, "proxy.config.hostdb.storage_path", sizeof(storage_path));
-  REC_ReadConfigInt32(storage_size, "proxy.config.hostdb.storage_size");
+  REC_ReadConfigString(hostdb_filename, "proxy.config.hostdb.filename", sizeof(hostdb_filename));
+
+  // Max number of items
+  REC_ReadConfigInt32(hostdb_max_count, "proxy.config.hostdb.max_count");
+  // max size allowed to use
+  REC_ReadConfigInt32(hostdb_max_size, "proxy.config.hostdb.max_size");
```

Comment:
Well, these are actually set in the default config options. So i'll set the default here in the code to -1, but either way the config option will overwrite it.